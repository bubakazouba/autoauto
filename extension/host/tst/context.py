import os
import sys
sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))

import adhoc2
import patternfinder
import keygrouper