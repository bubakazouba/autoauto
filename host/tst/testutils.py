import os
import json

def getFilePath(fileName):
    relativeFilePath = "./testdata/" + fileName
    scriptDir = os.path.dirname(__file__)
    return os.path.join(scriptDir, relativeFilePath)

def jsonLoad(fileName):
    file = open(getFilePath(fileName), "r")
    return byteify(json.load(file))
