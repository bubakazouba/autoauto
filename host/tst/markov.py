import unittest
from context import markov
import testutils

class TestMarkovMethods(unittest.TestCase):
    def markov_chain_test(self):
        cases = testutils.jsonLoad("markov_test_cases.json")
        for idx, case in enumerate(cases):
            chain = MarkovChain()
            for action1, action2 in zip(case[:-1], case[1:]):
                chain.add_transition(action1['key'], action2['key'])
            chain.resolve_transition_probabilities()
            chain.save("case_{}".format(idx))

if __name__ == '__main__':
    unittest.main()