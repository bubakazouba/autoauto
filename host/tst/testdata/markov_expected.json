[
    [
    {
        "key": "1",
        "tab": 1
    },
    {
        "key": "Enter",
        "tab": 1
    },
    {
        "key": "2",
        "tab": 1
    },
    {
        "key": "Enter",
        "tab": 1
    },
    {
        "key": "3",
        "tab": 1
    },
    {
        "key": "Enter",
        "tab": 1
    },
    {
        "key": "ArrowRight",
        "tab": 1
    },
    {
        "key": "cmd+ArrowUp",
        "tab": 1
    },
    {
        "key": "a",
        "tab": 1
    },
    {
        "key": "Enter",
        "tab": 1
    },
    {
        "key": "b",
        "tab": 1
    },
    {
        "key": "Enter",
        "tab": 1
    },
    {
        "key": "c",
        "tab": 1
    },
    {
        "key": "Enter",
        "tab": 1
    },
    {
        "key": "ArrowRight",
        "tab": 1
    },
    {
        "key": "cmd+ArrowUp",
        "tab": 1
    }],
    [
    {
        "key": "shift+cmd+ArrowDown",
        "tab": 1
    },
    {
        "key": "cmd+c",
        "tab": 1
    },
    {
        "key": "ArrowRight",
        "tab": 1
    },
    {
        "key": "cmd+v",
        "tab": 1
    },
    {
        "key": "cmd+ArrowLeft",
        "tab": 1
    },
    {
        "key": "cmd+ArrowDown",
        "tab": 1
    },
    {
        "key": "cmd+ArrowDown",
        "tab": 1
    },
    {
        "key": "shift+cmd+ArrowDown",
        "tab": 1
    }]
]